crates/recursor/src/error.rs

```diff
@@ -198,6 +240,28 @@ impl Clone for ErrorKind {
     }
 }
 
+impl From<Error> for ProtoError {
```

Comment:
This is used when translating recursor errors back to proto errors, which the lower level request handlers expect for error conditions.  Rationale for making this as a from implementation vs doing inline:

1) I think it is likely there are other error cases that are not being handled correctly today and that are being masked by the string error conversion currently in use. In particular, there are some cases where we return NoError today where we should be returning ServFail. I haven't run those to ground yet, but I think it is likely to be another error conversion problem. It would be nice to fix those in one place.

2) In the future, it would be nice to convert the other places in the recursor where error responses are generated (build_forwarded_response in particular) to use this and handle error response generation consistently.

Reply:
> This is used when translating recursor errors back to proto errors, which the lower level request handlers expect for error conditions.

This is the part I was wondering about. Feel free to merge this PR in the meantime, but this sounds like a design that might get us in trouble so I'd like to better understand how/where this happens.